### PR TITLE
move getCredential to verifiable pkg and add proof core claim verification

### PR DESCRIPTION
**Pull request by volodymyr-basiuk**

### CODE REVIEW

File: processor/processor.go [hunk 1]
```diff
@@ -25,20 +24,15 @@ type Validator interface {
 
 // Parser is an interface to parse claim slots
 type Parser interface {
+	// Deprecated: use credential.GetClaim instead
```
Comment (vmidyllic) [inline processor/processor.go:27]: ToCoreClaim

File: verifiable/credential.go [hunk 2]
```diff
@@ -60,6 +61,11 @@ func (vc *W3CCredential) VerifyProof(ctx context.Context, proofType ProofType,
 		return errors.New("can't get core claim")
 	}
 
+	err = vc.verifyCredentialClaim(ctx, coreClaim, verifyConfig.merklizeOptions)
```
Comment (vmidyllic) [inline verifiable/credential.go:64]: verifyCredentialCoreClaim

File: verifiable/credential.go [hunk 3]
```diff
@@ -82,6 +88,66 @@ func (vc *W3CCredential) VerifyProof(ctx context.Context, proofType ProofType,
 	}
 }
 
+func (vc *W3CCredential) verifyCredentialClaim(ctx context.Context, proofCoreClaim *core.Claim, merklizeOptions []merklize.MerklizeOption) error {
```
Comment (vmidyllic) [inline verifiable/credential.go:91]: verifyCredentialCoreClaim

File: verifiable/credential.go (hunk 2 above)
Comment (volodymyr-basiuk) [reply verifiable/credential.go]: done

File: processor/processor.go (hunk 1 above)
Comment (volodymyr-basiuk) [reply processor/processor.go]: done

**Pull request merged**